### PR TITLE
[BACKLOG-23575] Pentaho App Builder is broken due to invalid bean id call

### PR DESCRIPTION
**Pull request by davidmsantos90**
@webdetails/millenniumfalcon please review

### CODE REVIEW

**Comment by davidmsantos90** [comment]
@webdetails/millenniumfalcon @graimundo PR updated. please review

File: pentaho5/src/main/java/pt/webdetails/cpk/InterPluginBroker.java [hunk 1]
```diff
@@ -45,49 +47,63 @@ public static void run( Map<String, Object> params, OutputStream out ) throws Ex
...
 
     return null;
   }
 
-  private static boolean beanExists( IPluginCall pluginCall ) {
-    return ( (InterPluginCall) pluginCall).beanExists();
+  private static String getSettingValue( String tag, String defaultValue ) {
+    final PluginSettings settings = PluginEnvironment.env().getPluginSettings();
+
+    String value = settings.getTagValue( tag ).get( 0 );
```
Comment (graimundo) [inline pentaho5/src/main/java/pt/webdetails/cpk/InterPluginBroker.java]: I think we get an IndexOutOfBoundException here. You probably want to check  something lke this:
```
List<String> values = settings.getTagValue(tag);
return values.isEmpty() ? defaultValue : values.get(0);
```

File: pentaho5/src/main/java/pt/webdetails/cpk/InterPluginBroker.java [hunk 2]
```diff
@@ -33,7 +35,7 @@
   private static final String CDE_RENDER_API_RENDER_METHOD_TAG = "cde-render-api-render-method";
   private static final String CDE_RENDER_API_RENDER_METHOD = "render";
 
-  private static IPluginCall cdeRenderApiCall;
+  private static IPluginCall pluginCall;
```
Comment (graimundo) [inline pentaho5/src/main/java/pt/webdetails/cpk/InterPluginBroker.java]: I liked the first name better, `cdeRenderApiCall` or `cdeRenderCall`. Its more explicti

File: pentaho5/src/main/java/pt/webdetails/cpk/InterPluginBroker.java [hunk 3]
```diff
@@ -45,11 +48,18 @@ public static void run( Map<String, Object> params, OutputStream out ) throws Ex
...
     IPluginCall pluginCall = getPluginCall();
+    if ( pluginCall == null ) {
+      final String pluginId = CorePlugin.CDE.getId();
+      final String beanId = getSettingValue( CDE_RENDER_API_BEAN_ID_TAG, CDE_RENDER_API_BEAN_ID );
+      final String method = getSettingValue( CDE_RENDER_API_RENDER_METHOD_TAG, CDE_RENDER_API_RENDER_METHOD );
 
-    String response = pluginCall != null
-      ? pluginCall.call( parameters.getParameters() )
-      : "No valid InterPluginCall available";
+      logger.error( "No valid InterPluginCall available" +
```
Comment (graimundo) [inline pentaho5/src/main/java/pt/webdetails/cpk/InterPluginBroker.java]: The log should be on each try of finding the bean or we shouldn't include information about the beanId and method on the log. I'll explain the reasoning better tomorrow in person.

**Pull request merged**